### PR TITLE
Port to Missings

### PR DESCRIPTION
**Pull request by nalimilan**
Improve consistency of errors thrown by `recode()` by always throwing a `MissingException` when encountering a non-recoded missing value and `dest` does not allow them (rather than a `MethodError` for the `AbstractArray` method).

This will only pass CI when Missings.jl is tagged.

### CODE REVIEW

File: docs/src/using.md [hunk 1]
```diff
@@ -163,40 +163,40 @@ julia> y[1]
...
+4-element CategoricalArrays.CategoricalArray{Union{Missings.Missing, String},1,UInt32}:
  "Old"   
  "Young" 
  "Middle"
  "Young" 
 
-julia> levels!(y, ["Young", "Middle"]; nullok=true)
-4-element CategoricalArrays.CategoricalArray{Union{Nulls.Null, String},1,UInt32}:
- null    
+julia> levels!(y, ["Young", "Middle"]; missingok=true)
```
Comment (nalimilan) [inline docs/src/using.md]: We should find a better name for that argument, `nullok` wasn't great either. Maybe `allow_missing=true`?

File: src/CategoricalArrays.jl [hunk 2]
```diff
@@ -23,7 +23,7 @@ module CategoricalArrays
     include("value.jl")
 
     include("array.jl")
-    include("nullablearray.jl")
+    include("missingarray.jl")
```
Comment (nalimilan) [inline src/CategoricalArrays.jl]: "missingarray" isn't a great name, but I couldn't find anything better ("missingable" is ugly). Anyway it's internal.

File: docs/src/implementation.md [hunk 3]
```diff
@@ -2,20 +2,20 @@
...
  - `N` is the number of array dimensions.
  - `R` is the reference type, the element type of the `refs` field; it allows optimizing memory usage depending on the number of levels (i.e. `CategoricalArray` with less than 256 levels can use `R = UInt8`).
- - `V` is the type of the levels, it is equal to `T` for non-nullable arrays; for nullable arrays, `T = Union{V, Null}`
- - `C` is the type of categorical values, i.e. the objects returned when indexing non-null elements of `CategoricalArray`. `C` is `CategoricalString{R}` if `V = String` and `CategoricalValue{V, R}` for any other `V`.
- - `U` can be either `Union{}` for non-nullable arrays, or `Null` for the nullable ones.
+ - `V` is the type of the levels, it is equal to `T` for arrays which do not support missing values; for arrays which support missing values, `T = Union{V, Missing}`
+ - `C` is the type of categorical values, i.e. the objects returned when indexing non-missing elements of `CategoricalArray`. `C` is `CategoricalString{R}` if `V = String` and `CategoricalValue{V, R}` for any other `V`.
+ - `U` can be either `Union{}` for arrays which do not support missing values, or `Missing` for those who support them.
```
Comment (cjprybol) [inline docs/src/implementation.md]: `who support them` reads a little funny. Maybe use `those which` as you used above and below, or be more explicit and say `arrays which`?

Comment (nalimilan) [reply docs/src/implementation.md]: Good catch!

**Pull request merged**